src/store/playlist-store.js

```diff
@@ -1,12 +1,36 @@
-import { makeAutoObservable } from "mobx";
+import { makeAutoObservable, flow } from "mobx";
 
 class PlaylistStore {
-  myPlaylist = [];
-
   constructor(rootStore) {
     this.rootStore = rootStore;
 
-    makeAutoObservable(this);
+    this.myPlaylists = {};
+
+    makeAutoObservable(this, {
+      fetchMyPlaylists: flow,
+    });
+  }
+
+  setMyPlaylists(response) {
+    this.myPlaylists = response;
+  }
+
+  *fetchMyPlaylists(token) {
+    const response = yield fetch("https://api.spotify.com/v1/me/playlists", {
```

Comment:
"yield", for generator functions w/ the "*". follows the same pattern as "async await"